src/runtime_src/tools/xclbinutil/XclBinUtilities.cxx

```diff
@@ -909,7 +910,45 @@ transformMemoryBankGroupingCollections(const std::vector<boost::property_tree::p
   createMemoryBankGroupEntries(possibleGroupConnections, groupTopology, groupConnectivity);
 }
   
-
+bool
+XclBinUtilities::checkAIEPartitionIPLayoutCompliance(XclBin & xclbin){
+  // Get AIE_PARTITION metadata
+  std::set<std::string> allDpuKernelIDs;
+  Section *pAIEPartition = xclbin.findSection(AIE_PARTITION);
+  std::string jsonFile = pAIEPartition->getPathAndName();
+  boost::property_tree::ptree pt;
+  boost::property_tree::read_json(jsonFile, pt);
+  const boost::property_tree::ptree& ptAIEPartition = pt.get_child("aie_partition");
+  std::vector<boost::property_tree::ptree> ptPDIs = XUtil::as_vector<boost::property_tree::ptree>(ptAIEPartition, "PDIs");
+  for (const auto& pdi : ptPDIs) {
+    std::vector<boost::property_tree::ptree> ptCDOs = XUtil::as_vector<boost::property_tree::ptree>(pdi, "cdo_groups");
+    for (const auto& cdo_group : ptCDOs) {
+      std::vector<std::string> dpuKernelIDs = XUtil::as_vector_simple<std::string>(cdo_group, "dpu_kernel_ids");
+      allDpuKernelIDs.insert(dpuKernelIDs.begin(), dpuKernelIDs.end());
+    }
+  }
+                                             
+  // Get IP_LAYOUT metadata
```

Comment:
Please add a comment here to briefly describe what the DRC check is

Reply:
Added at the top.